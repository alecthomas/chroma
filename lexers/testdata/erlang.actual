-module(repl).

-export([run/0]).

run() ->
    read_eval_process().

read_eval_process() ->
    Line = io:get_line("> "),
    Out = process_line(Line),
    io:format("< ~s~n~n", [Out]),
    read_eval_process().

process_line(Line) ->
    string:uppercase(Line).
