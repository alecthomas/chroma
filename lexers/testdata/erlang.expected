[
  {"type":"Punctuation","value":"-"},
  {"type":"NameEntity","value":"module"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"repl"},
  {"type":"Punctuation","value":")."},
  {"type":"Text","value":"\n\n"},
  {"type":"Punctuation","value":"-"},
  {"type":"NameEntity","value":"export"},
  {"type":"Punctuation","value":"(["},
  {"type":"Name","value":"run"},
  {"type":"Operator","value":"/"},
  {"type":"LiteralNumberInteger","value":"0"},
  {"type":"Punctuation","value":"])."},
  {"type":"Text","value":"\n\n"},
  {"type":"NameFunction","value":"run"},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"-\u003e"},
  {"type":"Text","value":"\n    "},
  {"type":"Name","value":"read_eval_process"},
  {"type":"Punctuation","value":"()."},
  {"type":"Text","value":"\n\n"},
  {"type":"NameFunction","value":"read_eval_process"},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"-\u003e"},
  {"type":"Text","value":"\n    "},
  {"type":"NameVariable","value":"Line"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"NameNamespace","value":"io"},
  {"type":"Punctuation","value":":"},
  {"type":"NameFunction","value":"get_line"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralString","value":"\"\u003e \""},
  {"type":"Punctuation","value":"),"},
  {"type":"Text","value":"\n    "},
  {"type":"NameVariable","value":"Out"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"process_line"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"Line"},
  {"type":"Punctuation","value":"),"},
  {"type":"Text","value":"\n    "},
  {"type":"NameNamespace","value":"io"},
  {"type":"Punctuation","value":":"},
  {"type":"NameFunction","value":"format"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralString","value":"\"\u003c "},
  {"type":"LiteralStringInterpol","value":"~s~n~n"},
  {"type":"LiteralString","value":"\""},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"["},
  {"type":"NameVariable","value":"Out"},
  {"type":"Punctuation","value":"]),"},
  {"type":"Text","value":"\n    "},
  {"type":"Name","value":"read_eval_process"},
  {"type":"Punctuation","value":"()."},
  {"type":"Text","value":"\n\n"},
  {"type":"NameFunction","value":"process_line"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"Line"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"-\u003e"},
  {"type":"Text","value":"\n    "},
  {"type":"NameNamespace","value":"string"},
  {"type":"Punctuation","value":":"},
  {"type":"NameFunction","value":"uppercase"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"Line"},
  {"type":"Punctuation","value":")."},
  {"type":"Text","value":"\n"}
]
